packages/trader/src/Modules/Trading/Components/Form/TradeParams/Turbos/barrier-selector.jsx

```diff
@@ -3,14 +3,53 @@ import React from 'react';
 import { Localize, localize } from '@deriv/translations';
 import { connect } from 'Stores/connect';
 import { CSSTransition } from 'react-transition-group';
-import { Icon, Text, ThemedScrollbars } from '@deriv/components';
+import { Button, Icon, MobileDialog, Text, ThemedScrollbars, Popover } from '@deriv/components';
 import { isMobile } from '@deriv/shared';
 import Fieldset from 'App/Components/Form/fieldset.jsx';
 import BarriersList from './barriers-list.jsx';
 
 const BarrierSelector = ({ barrier_1, onChange, setHoveredBarrier, turbos_barrier_choices }) => {
     const [selected_barrier, setSelectedBarrier] = React.useState(barrier_1);
```

Comment:
As far as we have multiple useState, maybe it'll be better to apply [useReducer](https://beta.reactjs.org/reference/react/useReducer) for managing states?